File: State/cpp/OFF.cpp
#include "OFF.h"
#include "ON.h"

OFF::OFF(){
	cout << "   OFF-ctor ";
}

OFF::~OFF(){
	cout << "   dtor-OFF\n";
	//delete this;
}

void OFF::on(Machine *m){
	cout << "   going from OFF to ON";
	m->setCurrent(new ON());
	
}

File: State/cpp/java/src/dp/state/sm/OFF.java
package dp.state.sm;

public class OFF extends State {
	
	// Constructor
	public OFF(){
		super();
		System.out.println("   OFF-constructor ");
	}
	
	public void on(Machine m){
		System.out.println("   going from OFF to ON");
		m.setCurrent(new ON());
	}

}